spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/OpenTelemetryConfigurations.java

```diff
@@ -168,6 +187,139 @@ OtelHttpServerHandler otelHttpServerHandler(OpenTelemetry openTelemetry) {
 					new DefaultHttpServerAttributesExtractor());
 		}
 
+		@Configuration(proxyBeanMethods = false)
+		static class PropagationConfiguration {
+
+			@Bean
+			@ConditionalOnMissingBean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "B3", matchIfMissing = true)
+			TextMapPropagator b3TextMapPropagator() {
+				return B3Propagator.injectingSingleHeader();
+			}
+
+			@Configuration(proxyBeanMethods = false)
+			@ConditionalOnProperty(value = "management.tracing.baggage.enabled", havingValue = "false")
+			static class NoBaggagePropagatorConfiguration {
+				@Bean
+				@ConditionalOnMissingBean
+				@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "W3C")
+				TextMapPropagator w3cTextMapPropagator() {
+					return W3CTraceContextPropagator.getInstance();
+				}
+
+			}
+
+			@Bean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "AWS")
+			@ConditionalOnClass(AWSPropagation.class)
+			TextMapPropagator awsTextMapPropagator() {
+				return AwsXrayPropagator.getInstance();
+			}
+
+			@Configuration(proxyBeanMethods = false)
+			@ConditionalOnProperty(value = "management.tracing.baggage.enabled", matchIfMissing = true)
+			static class BaggagePropagatorConfiguration {
+
+				@Bean
+				@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "B3")
+				TextMapPropagator baggageTextMapPropagator(TracingProperties properties, BaggageManager baggageManager) {
+					return new BaggageTextMapPropagator(properties.getBaggage().getRemoteFields(), baggageManager);
+				}
+
+				@Bean
+				@ConditionalOnMissingBean
+				@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "W3C")
+				TextMapPropagator w3cTextMapPropagator() {
+					return TextMapPropagator.composite(W3CTraceContextPropagator.getInstance(), W3CBaggagePropagator.getInstance());
+				}
+
+				@Bean
+				@Conditional(BaggageTagSpanHandlerCondition.class)
+				BaggageTaggingSpanProcessor baggageTaggingSpanProcessor(TracingProperties properties) {
+					return new BaggageTaggingSpanProcessor(properties.getBaggage().getTagFields());
+				}
+
+				/**
+				 * We need a special condition as it users could use either comma or yaml encoding,
+				 * possibly with a deprecated prefix.
```

Comment:
Why could there be a deprecated prefix?

Reply:
Most likely that's some ancient thing that I unnecessarily copied over from Sleuth. I'll remove it